app/hooks/useAppAuth.ts

```
@@ -0,0 +1,15 @@
+import { useAuth } from "@workos-inc/authkit-nextjs/components";
+import { isWorkOSEnabled } from "@/lib/auth-utils";
+
+/**
+ * Custom hook that handles authentication for both WorkOS and anonymous modes
+ * Returns consistent user/loading state regardless of auth mode
+ */
+export const useAppAuth = () => {
+  if (isWorkOSEnabled()) {
+    return useAuth();
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)